src/arch/x86/reg/amd64_core.rs

```diff
@@ -0,0 +1,155 @@
+use crate::arch::Registers;
+use core::convert::TryInto;
+
+/// 64-bit x86 core registers.
+///
+/// Source: https://github.com/bminor/binutils-gdb/blob/master/gdb/features/arm/arm-core.xml
+#[derive(Default)]
+pub struct Amd64CoreRegs {
+    // RAX, RBX, RCX, RDX, RSI, RDI, RBP, RSP, r8-r15
+    regs: [u64; 16],
+    // Status register
+    eflags: u32,
+    // Instruction pointer
+    rip: u64,
+    // Segment registers: CS, SS, DS, ES, FS, GS
+    segments: [u32; 6],
+    // FPU registers: ST0 through ST7
+    st_regs: [F80; 8],
+    // SIMD Registers: XMM0 through XMM15
+    xmm_regs: [u128; 0x10],
+    // SSE Status/Control Register
+    mxcsr: u32,
+}
+
+type F80 = [u8; 10];
+
+impl Registers for Amd64CoreRegs {
+    fn gdb_serialize(&self, mut write_byte: impl FnMut(Option<u8>)) {
+        macro_rules! write_bytes {
+            ($bytes:expr) => {
+                for b in $bytes {
+                    write_byte(Some(*b))
+                }
+            };
+        }
+
+        for reg in &self.regs {
+            write_bytes!(&reg.to_le_bytes());
+        }
+
+        // rip
+        write_bytes!(&self.rip.to_le_bytes());
+
+        // eflags
+        write_bytes!(&self.eflags.to_le_bytes());
+
+        // cs, ss, ds, es, fs, gs
+        for seg in &self.segments {
+            write_bytes!(&seg.to_le_bytes());
+        }
+
+        // st0 to st7
+        for st_reg in &self.st_regs {
+            write_bytes!(st_reg);
+        }
+
+        // fctrl, fstat, ftag, fiseg, fioff, foseg, fooff, fop
+        for _ in 0..8 {
+            write_bytes!(&[0; 4]);
```

Comment:
I'm not too familiar with x86 innards, are these registers relatively unused nowadays? If so, I don't mind leaving them stubbed out, though if it's not too much trouble, [de]serializing them would be nice.

Reply:
Ok so I took a second stab at those because GDB has very non-standard names for them on top of them basically being internal registers for the x87 FPU (which virtually nobody really uses anyway) it was rather difficult, but after re-reading all of the FPU section of the intel manual on it, I managed to figure out what the heck was up with the GDB naming scheme. Should be sane now.

Reply:
Ah, nice sleuthing!